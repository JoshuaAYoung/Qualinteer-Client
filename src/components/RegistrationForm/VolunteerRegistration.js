import React, { Component } from 'react';

export default class VolunteerRegistration extends Component {
  render() {
    return (
      <div>
        <h2 className='formHeader'>Sign up to offer help</h2>
        <div>
          <label htmlFor='email'>Email Address</label>
          <input placeholder='Email' type='text' name='email' id='email' />
        </div>
        <div>
          <label htmlFor='password'>Password</label>
          <input placeholder='Password' type='password' name='password' id='password' />
        </div>
        <div>
          <label htmlFor='full-name'>Full name</label>
          <input placeholder='Full Name' type='text' name='name' id='name' />
        </div>
        <div>
          <label htmlFor='phone'>Phone Number (optional)</label>
          <input placeholder='Phone Number' type='text' name='phone' id='phone' />
        </div>
      </div >
    )
  }
}