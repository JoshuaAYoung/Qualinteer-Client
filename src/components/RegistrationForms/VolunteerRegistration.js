import React, { Component } from 'react';
import UserFields from './UserFields'

export default class VolunteerRegistration extends Component {
  render() {
    return (
      <div>
        <h2 className='pageHeader'>Sign up to volunteer</h2>
        <UserFields
          addTempValidation={this.props.addTempValidation}
          addTempValue={this.props.addTempValue}
          errorProps={this.props.errorProps}
        />
      </div >
    )
  }
}