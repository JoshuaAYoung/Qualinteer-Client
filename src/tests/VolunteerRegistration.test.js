import React from 'react';
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import VolunteerRegistration from '../components/RegistrationForms/VolunteerRegistration'

describe('VolunteerRegistration Component', () => {
  const props = {
    errorProps: {
      emailError: 'test error 1',
      passwordError1: 'test error 1',
      passwordError2: 'test error 1',
      passwordError3: 'test error 1',
      passwordError4: 'test error 1'
    }
  }

  it('VolunteerRegistration Smoke Test', () => {
    const div = document.createElement('div')

    ReactDOM.render(<VolunteerRegistration {...props} />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('VolunteerRegistration Snapshot Test', () => {
    const wrapper = shallow(
      <VolunteerRegistration {...props} />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})