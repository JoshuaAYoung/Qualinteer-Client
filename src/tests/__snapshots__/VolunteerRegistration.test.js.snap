// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VolunteerRegistration Component VolunteerRegistration Snapshot Test 1`] = `
<div>
  <UserFields
    errorProps={
      Object {
        "emailError": "test error 1",
        "passwordError1": "test error 1",
        "passwordError2": "test error 1",
        "passwordError3": "test error 1",
        "passwordError4": "test error 1",
      }
    }
  />
</div>
`;
